back/src/index.js

```diff
@@ -119,6 +121,59 @@ app.get('/comments', (_, res) => {
   });
 });
 
+app.post('/auth', (req, res) => {
+  console.log(req.body);
+  db.User.findOne({ where: { email: req.body.email } }).then((user) => {
+    if (user && comparePassword(req.body.password, user.password)) {
+      const data = {
+        nickname: user.nickname,
+        email: user.email,
+        authLevel: user.authLevel,
+        biography: user.biography,
+        icon: user.icon,
+        enabled: user.enabled,
+      };
+      jsonwebtoken.sign(data, 'privateKey', { expiresIn: '1h' },
+        (err, token) => {
+          if (err) { console.log(err); }
+          res.send({ token });
+        });
+    } else {
+      res.sendStatus(401);
+    }
+  });
+});
+
+app.post('/register', (req, res) => {
+  console.log(req.body);
+  db.User.findOne({ where: { email: req.body.email } }).then((user) => {
+    if (user) {
+      res.sendStatus(500);
+    } else {
+      const newUser = db.User.create({
+        nickname: req.body.nickname,
+        email: req.body.email,
+        biography: '',
+        authLevel: 2,
+        password: hashPassword(req.body.password),
+      });
+      const data = {
+        nickname: newUser.nickname,
+        email: newUser.email,
+        authLevel: newUser.authLevel,
+        biography: newUser.biography,
+        icon: newUser.icon,
+        enabled: newUser.enabled,
+      };
+      jsonwebtoken.sign(data, 'privateKey', { expiresIn: '1h' },
+        (err, token) => {
+          if (err) { console.log(err); }
```

Comment:
`if (err) { console.log(err); return;}`
Pour éviter de renvoyer le token même en cas d'erreur au moment de la signature

Reply:
Resolved in 2131c8dc98da6ec64382e2a1e66af6b64050ae9c